Silabs lighting-app build failing in Jenkins CI due to interactive SLT “Preferred SLT” prompt (BRD4187C) leading to version-field read failure

## Summary
Silabs build for `examples/lighting-app/silabs` is failing in Jenkins CI when targeting **BRD4187C**. The failure is triggered by an **interactive SLT prompt** introduced/encountered during the build:

> “Preferred SLT… There is already a preferred slt placed at … Do you want to replace it and run this one instead [y/n]?”

Because Jenkins CI is non-interactive, the build step cannot reliably answer this prompt, and subsequent SLT/package steps end up failing with metadata parsing errors:

- `Could not read required version fields: sdk_version from .../simplicity_sdk.slcs`
- `Could not read required version fields: version from .../wiseconnect.slce`

This blocks `lighting-app` builds in CI.

## Environment
- Date observed: **2026-03-09** (from logs: `20260309 01:25:26`)
- OS: Linux
- CI: Jenkins (`Production_Ready_Daily_Stress_Test_Runs`)
- Target board: **BRD4187C**
- App: `lighting-app`
- Example path: `examples/lighting-app/silabs`
- Script: `./scripts/examples/gn_silabs_example.sh`

## What changed / suspected root cause
The build now triggers an SLT “Preferred SLT” selection flow that requires user input (`y/n`). This is not compatible with Jenkins CI and causes the SLT install/update and subsequent package steps to proceed in an inconsistent state (or with mixed SLT installs), ultimately resulting in failure to read required version fields from:
- `third_party/silabs/simplicity_sdk/simplicity_sdk.slcs` (`sdk_version`)
- `third_party/silabs/wifi_sdk/wiseconnect.slce` (`version`)

## Evidence from logs
Repeated interactive prompt during CI:
- `************ Preferred SLT ************`
- `SLT is designed to only run one instance on each host.`
- `There is already a preferred slt placed at .../Runs/816/.../scripts/setup/silabs/slt.`
- `Do you want to replace it and run this one instead [y/n]?`

Final failure:
- `[ERROR] Could not read required version fields: sdk_version from .../simplicity_sdk.slcs; version from .../wiseconnect.slce`
- `Build failed for lighting-app`


## Attachments
- Full Jenkins console log (attached)

[Silabs_build_failure_log.txt](https://github.com/user-attachments/files/25899517/Silabs_build_failure_log.txt)

--- a/scripts/setup/silabs/install-packages.py
+++ b/scripts/setup/silabs/install-packages.py
@@ -224,7 +224,7 @@ def update_slt_cli(slt_cli_path):
     update_cmd = [slt_cli_path, "update", "--self"]
     try:
         logger.info("Updating SLT CLI to latest version...")
-        subprocess.run(update_cmd, check=True)
+        subprocess.run(update_cmd, stdin=subprocess.DEVNULL, check=True)
         logger.info("SLT CLI updated successfully")
     except subprocess.CalledProcessError as e:
         logger.warning("Failed to update slt-cli: %s", e)
@@ -255,7 +255,7 @@ def install_sdk_packages(slt_cli_path):
         install_cmd = [slt_cli_path, "install", "-f", pkg_path]
         try:
             logger.info("Installing packages from %s...", os.path.basename(pkg_path))
-            subprocess.run(install_cmd, check=True)
+            subprocess.run(install_cmd, stdin=subprocess.DEVNULL, check=True)
             logger.info("Packages from %s installed successfully", os.path.basename(pkg_path))
         except subprocess.CalledProcessError as e:
             logger.error("Failed to install packages from %s: %s", pkg_path, e)
@@ -267,17 +267,23 @@ def slt_where(slt_cli_path, package):
     try:
         result = subprocess.run(
             [slt_cli_path, "where", package],
+            stdin=subprocess.DEVNULL,
             capture_output=True,
             text=True,
             check=False,
         )
         if result.returncode == 0 and result.stdout.strip():
             # SLT may print interactive prompts to stdout before the actual path.
-            # Extract the last absolute path from the output. The path may appear
-            # on its own line or appended after a prompt (e.g. "[y/n]?/home/...").
-            path_match = re.findall(r"(/[^\s.,;:\"']+)", result.stdout)
-            if path_match:
-                return path_match[-1]
+            # The path may be on its own line or appended after a prompt
+            # (e.g. "[y/n]?/home/..."). Scan from the bottom to find it.
+            for line in reversed(result.stdout.strip().splitlines()):
+                line = line.strip()
+                if line.startswith("/"):
+                    return line
+                # Handle path concatenated after interactive prompt
+                prompt_idx = line.rfind("?/")
+                if prompt_idx != -1:
+                    return line[prompt_idx + 1:]
             logger.error(
                 "Could not find an absolute path in 'slt where %s' output:\n%s",
                 package, result.stdout.strip(),
@@ -380,8 +386,11 @@ def create_sdk_symlinks(simplicity_sdk_path, wiseconnect_path):
 
     def create_symlink(target_path, link_name):
         if not target_path or not os.path.isdir(target_path):
-            logger.warning("Target path does not exist or is not a directory: %s", target_path)
-            return
+            logger.error(
+                "Cannot create symlink: target path does not exist or is not a directory: %s",
+                target_path,
+            )
+            sys.exit(1)
         link_path = os.path.join(silabs_dir, link_name)
         try:
             os.makedirs(silabs_dir, exist_ok=True)
@@ -390,15 +399,20 @@ def create_sdk_symlinks(simplicity_sdk_path, wiseconnect_path):
                     current = os.path.realpath(link_path)
                     if os.path.realpath(target_path) == current:
                         logger.info("Symlink already up to date: %s", link_path)
-                        return
+                        return  # No error; skip create
                     os.remove(link_path)
                 else:
-                    logger.warning("Path exists and is not a symlink, skipping: %s", link_path)
-                    return
+                    logger.error(
+                        "Cannot create symlink: %s already exists and is not a symlink. "
+                        "Remove it manually and re-run.",
+                        link_path,
+                    )
+                    sys.exit(1)
             os.symlink(target_path, link_path)
             logger.info("Created symlink %s -> %s", link_path, target_path)
         except OSError as e:
-            logger.warning("Could not create symlink %s: %s", link_path, e)
+            logger.error("Could not create symlink %s -> %s: %s", link_path, target_path, e)
+            sys.exit(1)
 
     create_symlink(simplicity_sdk_path, "simplicity_sdk")
     create_symlink(wiseconnect_path, "wifi_sdk")

--- a/scripts/setup/silabs/install-packages.py
+++ b/scripts/setup/silabs/install-packages.py
@@ -280,10 +280,12 @@ def slt_where(slt_cli_path, package):
                 line = line.strip()
                 if line.startswith("/"):
                     return line
-                # Handle path concatenated after interactive prompt
-                prompt_idx = line.rfind("?/")
-                if prompt_idx != -1:
-                    return line[prompt_idx + 1:]
+                # Handle path concatenated after interactive prompt (e.g. "[y/n]?/path" or "[y/n]? /path")
+                q_idx = line.rfind("?")
+                if q_idx != -1:
+                    after = line[q_idx + 1:].lstrip()
+                    if after.startswith("/"):
+                        return after
             logger.error(
                 "Could not find an absolute path in 'slt where %s' output:\n%s",
                 package, result.stdout.strip(),

--- a/scripts/setup/silabs/install-packages.py
+++ b/scripts/setup/silabs/install-packages.py
@@ -272,9 +272,23 @@ def slt_where(slt_cli_path, package):
             check=False,
         )
         if result.returncode == 0 and result.stdout.strip():
-            return result.stdout.strip()
-    except (subprocess.SubprocessError, FileNotFoundError):
-        pass
+            # SLT may print interactive prompts to stdout before the actual path.
+            # Extract the last absolute path from the output. The path may appear
+            # on its own line or appended after a prompt (e.g. "[y/n]?/home/...").
+            path_match = re.findall(r"(/[^\s.,;:\"']+)", result.stdout)
+            if path_match:
+                return path_match[-1]
+            logger.error(
+                "Could not find an absolute path in 'slt where %s' output:\n%s",
+                package, result.stdout.strip(),
+            )
+        else:
+            logger.error(
+                "'slt where %s' failed (exit code %s): %s",
+                package, result.returncode, (result.stderr or result.stdout or "").strip(),
+            )
+    except (subprocess.SubprocessError, FileNotFoundError) as e:
+        logger.error("Failed to run 'slt where %s': %s", package, e)
     return None
 
 

